pkg/ui/src/views/statements/planView.tsx

```diff
@@ -16,115 +16,274 @@ import _ from "lodash";
+      }
+      return "nodeDetails expanded";
+    } else if (warn) {
+      return "nodeDetails warn";
     }
-  });
+    return "nodeDetails";
+  }
 
-  const collapsedAttrs = _.values(collapsed);
-  return _.sortBy(collapsedAttrs, (ca) => (
-    ca.key === "table" ? "table" : "z" + ca.key
-  ));
+  render() {
+    const node = this.props.node;
+    const header = planNodeHeaderProps(node);
+    return (
+      <div
+        className={this.renderClassName(header.warn)}
+        onClick={() => {
+          this.props.toggleExpanded();
+        }}
+      >
+        <PlanNodeDescription
+          header={header}
+        /> {this.renderArrow()}
+        {this.renderMaybeExpandedDetail()}
+      </div>
+    );
+  }
+}
+
+interface IPlanNodeProps {
+  node: IPlanNode;
+}
+
+interface IPlanNodeState {
+  expanded: boolean;
+}
+
+class PlanNode extends React.Component<IPlanNodeProps, IPlanNodeState> {
+  constructor(props: IPlanNodeProps) {
+    super(props);
+    this.state = {
+      expanded: false,
+    };
+
+    // This binding is necessary to make `this` work in the callback
+    this.toggleExpanded = this.toggleExpanded.bind(this);
+  }
+
+  // TODO(celia): toggleExpanded() was originally inside <PlanNodeDetails/>,
+  // but I've moved this here for now, because the outer <li> may need
+  // to know state.expanded in order to finish/fix up the styling of the
+  // node connecting lines. Should move this back into <PlanNodeDetails/>
+  // if <li> ends up not needing to know about expanded.
+  toggleExpanded() {
+    this.setState(state => ({
+      expanded: !state.expanded,
+    }));
+  }
+
+  renderClassName() {
+    const node = this.props.node;
+    const expanded = false; /* TODO */
+    if (node.attrs && node.attrs.length > 0 && expanded) {
+      return "expanded";
+    }
+  }
+
+  render() {
+    const node = this.props.node;
+    const expanded = this.state.expanded;
+    return <li className={this.renderClassName()}>
+      <PlanNodeDetails
+        node={node}
+        expanded={expanded}
+        toggleExpanded={this.toggleExpanded}
+      />
+      {node.children && node.children.map( (child) => (
+        <PlanNodes
+          nodes={child}
+        />
+      ))
+      }
+    </li>;
+  }
+}
+
+function PlanNodes(props: {
+  nodes: IPlanNode[],
+}): React.ReactElement<{}> {
+  const nodes = props.nodes;
+  return <ul>
+    {nodes.map( (node) => {
+      return <PlanNode node={node}/>;
+    })}
+  </ul>;
+}
+
+export function PlanView(props: {
+  title: string,
+  plan: IExplainTreePlanNode,
+}) {
+  const flattenedPlanNodes = flattenTree(props.plan);
+  return <table className="plan-view-table">
+    <thead>
+    <tr className="plan-view-table__row--header">
+      <th className="plan-view-table__cell">
+        {props.title}
+        <div className="plan-view-table__tooltip">
+          <ToolTipWrapper
+            text="This is the most recently sampled plan. Plans are sampled every five minutes.">
```

Comment:
A little confused about how this works… If a query of this fingerprint hasn't occurred in the last five minutes, will there still be one here?

Reply:
cc @Amruta-Ranade; we'll def have to explain this in docs…

Reply:
Hmm, how this works under the hood: Every time a statement is run, if the time since the last sample was taken is greater than 5 minutes[1], then we sample a new one.

@piyush-singh @Amruta-Ranade:

1. Thoughts on copy to better explain the above?
2. The "5 minutes" can be changed in cluster settings (`sql.metrics.statement_details.plan_collection.period`). So if a user changes this value, that means this copy will no longer be accurate. So maybe this tooltip should be similar tooltip on the Statements page:

![image](https://user-images.githubusercontent.com/3051672/52236687-84841e80-2895-11e9-8e58-42787ada470a.png)

(Note that we have the `time last sampled` for each logical plan, so we can do something similar to the above screenshot.)